types/usb/index.d.ts

```diff
@@ -55,14 +58,18 @@ export class ConfigDescriptor {
   bmAttributes: number;
   bMaxPower: number;
   extra: Buffer;
+  interfaces: InterfaceDescriptor[][];
 }
 
 export class Interface {
+  interface: number;
+  altSetting: number;
   descriptor: InterfaceDescriptor;
   endpoints: Endpoint[];
   constructor(device: Device, id: number);
   claim(): void;
-  release(closeEndpoints?: (err?: string) => void, cb?: (err?: string) => void): void;
+  release(cb?: (err?: string) => void): void;
```

Comment:
I think this still doesn't support the situation where `closeEndpoints` is passed in as a `Function`

Reply:
`closeEndpoints` is never a function, the code supports a situation where `closeEndpoints` is not passed in and the `cb` callback function is the first parameter. From the `node-usb` code:

```javascript
	if (typeof closeEndpoints == 'function') {
		cb = closeEndpoints;
		closeEndpoints = null;
	}
```

Hence why there are two definitions for `release()`:

```javascript
    release(closeEndpoints?: (err?: string) => void, cb?: (err?: string) => void): void;
    release(cb?: (err?: string) => void): void;
```

Reply:
Oh i see, I didn't read that carefully enough (stopped at `if (typeof closeEndpoints == 'function')` 🤦‍♂️ ). That's what I get for reviewing on my phone. Sorry!

Reply:
👍